src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -7336,25 +7363,92 @@ and TcInterpolatedStringExpr cenv (overallTy: OverallTy) env m tpenv (parts: Syn
             // Type check the expressions filling the holes
             let fillExprs, tpenv = TcExprsNoFlexes cenv env m tpenv argTys synFillExprs
 
-            let fillExprsBoxed = (argTys, fillExprs) ||> List.map2 (mkCallBox g m)
-
-            let argsExpr = mkArray (g.obj_ty, fillExprsBoxed, m)
-            let percentATysExpr =
-                if percentATys.Length = 0 then
-                    mkNull m (mkArrayType g g.system_Type_ty)
+            // If all fill expressions are strings and there is less then 5 parts of the interpolated string total
+            // then we can use System.String.Concat instead of a sprintf call.
+            // First filter out all string parts that are empty or only have a '%s' string specifier,
+            // also get rid of the '%s' specifier right before the interpolation expression.
+            let nonEmptyParts =
```

Comment:
Maybe could be turned into a function to not evaluate this when the feature is not enabled for example.

Reply:
I decided to use another suggestion how to refactor and it solves this issue as well